pkg/types/chain_writer.go

```diff
@@ -7,12 +7,94 @@ import (
 	"github.com/google/uuid"
 )
 
+// CallOption can be used to configure a single call to the ChainWriter.
+type CallOption interface {
+	apply(*CallOptions)
+}
+
+// CallOptionFunc is a functional type that implements the CallOption interface.
+type callOptionFunc func(*CallOptions)
+
+func (fn callOptionFunc) apply(o *CallOptions) {
+	fn(o)
+}
+
+// CallOptions contains the values which are configurable per-call to the ChainWriter.
+//
+// ChainWriter implementations should hold an instance of this struct and apply the options to the struct before making the call.
+type CallOptions struct {
+	// TransactionID is submitted to the particular chain's TXM as an idempotency key.
+	// It should be unique across transaction.
+	TransactionID   string
+	TransactionMeta *TxMeta
+	Value           *big.Int
+}
+
+// Apply applies the given options to the CallOptions.
+func (o *CallOptions) Apply(opts ...CallOption) {
+	for _, opt := range opts {
+		opt.apply(o)
+	}
+}
+
+// WithTransactionID sets the transactionID (idempotency key) for a call to the ChainWriter.
+//
+// Default: ""
+func WithTransactionID(transactionID string) CallOption {
+	return callOptionFunc(func(o *CallOptions) {
+		o.TransactionID = transactionID
+	})
+}
+
+// WithTransactionMeta sets the transaction metadata for a call to the ChainWriter.
+//
+// Default: <nil>
+func WithTransactionMeta(meta *TxMeta) CallOption {
+	return callOptionFunc(func(o *CallOptions) {
+		o.TransactionMeta = meta
+	})
+}
+
+// WithValue sets the value to be sent with a transaction.
+//
+// Default: <nil>
```

Comment:
If Value is now an optional argument, does it make sense for it to be a pointer? We could just have it be an int with default value 0, right?

Reply:
It could be, but the canonical way to use `big.Int` types is as a pointer and we've had multiple people requesting the change to a pointer.